assets/js/base/context/cart-checkout/payment-methods/use-payment-method-registration.js

```diff
@@ -32,32 +52,38 @@ const usePaymentMethodRegistration = (
 	dispatcher,
 	registeredPaymentMethods
 ) => {
-	const { isEditor } = useEditorContext();
 	const [ isInitialized, setIsInitialized ] = useState( false );
 
 	/**
-	 * @type {Object} initializedMethodsDefault Used to hold payment methods that have been
-	 *                                          initialized.
+	 * @type {Object} initializedMethodsDefault Holds payment methods that have been initialized.
 	 */
 	const [
 		initializedPaymentMethods,
 		setInitializedPaymentMethods,
 	] = useState( {} );
+
+	const { isEditor } = useEditorContext();
 	const { shippingAddress } = useShippingDataContext();
 	const { cartTotals, cartNeedsShipping } = useStoreCart();
 	const canPayArgument = useRef( {
 		cartTotals,
 		cartNeedsShipping,
 		shippingAddress,
 	} );
-	const countPaymentMethodsInitializing = useRef(
-		Object.keys( registeredPaymentMethods ).length
-	);
 
-	const setInitializedPaymentMethod = ( paymentMethod ) => {
+	/**
+	 * Update initialized payment methods in local state.
+	 *
+	 * @param {Object} paymentMethod Payment method object that has been initialized.
+	 * @param {boolean} enabled True if the method can be used, or false if payment is not possible via this method.
+	 */
+	const setInitializedPaymentMethod = ( paymentMethod, enabled = true ) => {
 		setInitializedPaymentMethods( ( paymentMethods ) => ( {
 			...paymentMethods,
-			[ paymentMethod.name ]: paymentMethod,
+			[ paymentMethod.name ]: {
+				enabled,
+				paymentMethod,
+			},
```

Comment:
I made this change so we can track initialization of methods even is canPay resolves as false. The list of methods is filtered later before dispatching.